tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,241 @@
+
+# make a list of all groups within the full tile file
+ls=h5ls(f,all=T)
+
+#make a list of all of the names within the Coordinate_System group
+cg=unique(ls[ls$group=="/SJER/Reflectance/Metadata/Coordinate_System",]$name)
+
+for(i in 1:length(cg)){
+  print(cg[i])
+  a=h5readAttributes(f,paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]))
+  d=h5read(f,paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]))
+  attributes(d)=a
+  h5write(obj=d,file=nf,
+          name=paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]),
+          write.attributes=T)
+}
+```
+
+## Spectral Subsetting
+
+The goal of subsetting this dataset is to substantially reduce the file size to make it faster to download and process these data. While each AOP mosaic tile is not particularly large in terms of its spatial scale (1km by 1km at 1m resolution= 1,000,000 pixels, or about half as many pixels at shown on a standard 1080p computer screen), the 426 spectral bands available result in a fairly large file size. While having hundreds of bands is important for advanced research, we can use far fewer bands for demonstartion and teaching purposes, while still getting a good sense of what these hyperspectral data can do. Therefore, we will reduce the spectral resolution of these data by selecting every fourth band in the dataset, which reduces the file size to 1/4 of the original!
+
+```{r subset-wavelengths}
+#First, we make out 'index'  a list of number that will allow us to select every fourth band
+idx <- seq(from = 1, to = 426, by = 4)
+
+# We then use this index to select particular wavelengths from the full tile using the "index=" argument
+ws <- h5read(file = f, 
+             name = "SJER/Reflectance/Metadata/Spectral_Data/Wavelength", 
+             index = list(idx)
+            )
+
+# As per above, we also need the wavelength attributes
+wa <- h5readAttributes(file = f, 
+                       name = "SJER/Reflectance/Metadata/Spectral_Data/Wavelength")
+attributes(ws) <- wa
+
+# Finally, write the subset of wavelengths and their attributes to the subset file
+h5write(obj=ws, file=nf,
+        name="SJER/Reflectance/Metadata/Spectral_Data/Wavelength",
+        write.attributes=T)
+```
+
+## Spatial Subsetting
+
+Even after spectral subsetting, our file size would be greater than 100Mb. Therefore, we will perform a spatial subsetting process as well when we read the hyperspectral imagery data. Note that the `index=` argument in `h5read` requires a `list()` object of three dimenstions for this example - in the order of band, column, row. Here, we are taking every fourth band (using our `idx` variabe), columns 501:1000 (the eastern half of the tile) and rows 1:500 (the northern half of the tile). The results in us extracting every fourth band of the northeast quadrant of the mosaic tile.
+
+
+```{r read-hyp-data}
+# Read in reflectance data - note the index argument! This is ultimately how we reduce the file size.
```

Comment:
1) Do you mean "index" argument, or the full list() one? 
2) Why are you doing this on "f" and not building off the spectral subsetting? I can see wanting to demonstrate both and not having them dependent on each other but describing this in the text would be good.

Reply:
1) ahh yes, I am trying to draw the reader's attention to the list that we are passing to the index= argument. I have clarified this in the latest commit.
2) Not sure I understand this comment. What we are doing here is extracting a subset of data from the 'f' file, saving that to a variable, adding attributes to that variable, then writing that variable to the new file 'nf' in the /Reflectance/Reflectance_Data group. In this way, the 'nf' is dependent upon the subset of the 'f' file as an input. I will leave this 'conversation' open pending your confirmation or further dispute.

Reply:
'f' in this tutorial is the full NEON tile, but 'f' in the hyperspectral series is the example subset. Would be better to change the name for this 'subset' tutorial so that the output is called 'f' to be consistent with the hyperspectral tutorial series.

Reply:
These changes have been completed in the latest commit